kernel/hashlib.h

```diff
@@ -277,11 +382,11 @@ class dict
 	}
 #endif
 
-	int do_hash(const K &key) const
+	Hasher::hash_t do_hash(const K &key) const
 	{
-		unsigned int hash = 0;
+		Hasher::hash_t hash = 0;
```

Comment:
This is the hashing state/result type rather than the Hasher struct that represents in-progress hashing, so I think it should be explicitly rather than implicitly zero initialized